### PR TITLE
Display store card checkbox based on session setup response

### PR DESCRIPTION
**Pull request by etman55**
## Description
- Display store card checkbox based on session setup response.

## Checklist 
- [x] Code is unit tested
- [x] Changes are tested manually
- [x] Add relevant labels to PR  

COAND-614

PS: update checkout api version to v70 to be able to test.

### CODE REVIEW

File: bcmc/src/main/java/com/adyen/checkout/bcmc/BcmcComponentProvider.kt
Comment (OscarSpruit) [inline bcmc/src/main/java/com/adyen/checkout/bcmc/BcmcComponentProvider.kt:53]: It's better to give this another name to avoid confusion in the `get()` where we pass the same thing. Maybe `dropInCheckoutSession` or `overrideCheckoutSession` is a good name 🤔

File: bcmc/src/test/java/com/adyen/checkout/bcmc/BcmcComponentParamsMapperTest.kt
Comment (OscarSpruit) [inline bcmc/src/test/java/com/adyen/checkout/bcmc/BcmcComponentParamsMapperTest.kt:95]: `ParameterizedTest`s would be perfect for this

File: drop-in/src/main/java/com/adyen/checkout/dropin/ComponentParsingProvider.kt
Comment (OscarSpruit) [inline drop-in/src/main/java/com/adyen/checkout/dropin/ComponentParsingProvider.kt:384]: It's nicer to have parameters with a default value at the end

File: drop-in/src/main/java/com/adyen/checkout/dropin/ui/viewmodel/DropInViewModel.kt
Comment (OscarSpruit) [inline drop-in/src/main/java/com/adyen/checkout/dropin/ui/viewmodel/DropInViewModel.kt:87]: This should use a custom getter, because the value in `bundleHandler` could change

File: example-app/src/main/java/com/adyen/checkout/example/data/api/model/SessionRequest.kt
Comment (OscarSpruit) [inline example-app/src/main/java/com/adyen/checkout/example/data/api/model/SessionRequest.kt:31]: What is this needed for? I don't see anything about it in the ticket

File: sessions-core/src/main/java/com/adyen/checkout/sessions/model/setup/Configuration.kt
Comment (OscarSpruit) [inline sessions-core/src/main/java/com/adyen/checkout/sessions/model/setup/Configuration.kt:10]: A more descriptive name would be nice as we already have 2 classes with this name. Maybe something like `SessionSetupConfiguration`

File: example-app/src/main/java/com/adyen/checkout/example/ui/card/SessionsCardTakenOverViewModel.kt
Comment (jreij) [inline example-app/src/main/java/com/adyen/checkout/example/ui/card/SessionsCardTakenOverViewModel.kt:119]: Instead of manually passing these values every time we need them, how about we add them as default values to the parameters of the `getSessionRequest` method?

File: sessions-core/src/main/java/com/adyen/checkout/sessions/model/setup/SessionSetupResponse.kt
Comment (jreij) [inline sessions-core/src/main/java/com/adyen/checkout/sessions/model/setup/SessionSetupResponse.kt:72]: You forgot to add this to the `serialize` function as well

File: drop-in/src/main/java/com/adyen/checkout/dropin/ComponentParsingProvider.kt
Comment (jreij) [inline drop-in/src/main/java/com/adyen/checkout/dropin/ComponentParsingProvider.kt:424]: This should also be done in the stored component

File: drop-in/src/main/java/com/adyen/checkout/dropin/ComponentParsingProvider.kt
Comment (jreij) [inline drop-in/src/main/java/com/adyen/checkout/dropin/ComponentParsingProvider.kt:404]: Let's pass it everywhere not just to card and bcmc, to keep the provider class consistent

File: drop-in/src/main/java/com/adyen/checkout/dropin/ui/viewmodel/DropInSavedStateHandleContainer.kt
Comment (jreij) [inline drop-in/src/main/java/com/adyen/checkout/dropin/ui/viewmodel/DropInSavedStateHandleContainer.kt:79]: Having both `checkoutSession` and `checkoutSession.sessionSetupResponse` is redundant, let's remove the previous

Comment (jreij) [reply drop-in/src/main/java/com/adyen/checkout/dropin/ui/viewmodel/DropInSavedStateHandleContainer.kt:79]: Or actually maybe we don't need to pass the whole `CheckoutSession` object to the providers. How about we only pass the `SessionSetupConfiguration` object, and we store it inside `SessionDetails` (just like the rest of the fields. Passing the whole session just to take one field seems overkill

File: bcmc/src/test/java/com/adyen/checkout/bcmc/BcmcComponentParamsMapperTest.kt
Comment (jreij) [inline bcmc/src/test/java/com/adyen/checkout/bcmc/BcmcComponentParamsMapperTest.kt:174]: This doesn't seem very readable, you can to check the main methods to know what each field is for. How about we create a data class with these 3 booleans and use it as a parameter for the test? This way we can used named arguments and we'll be able to read what each boolean is for

File: drop-in/src/main/java/com/adyen/checkout/dropin/ComponentParsingProvider.kt
Comment (etman55) [reply drop-in/src/main/java/com/adyen/checkout/dropin/ComponentParsingProvider.kt:424]: why do we need to do this? I mean stored component doesn't have that field.

File: drop-in/src/main/java/com/adyen/checkout/dropin/ComponentParsingProvider.kt
Comment (etman55) [reply drop-in/src/main/java/com/adyen/checkout/dropin/ComponentParsingProvider.kt:404]: If we passed this everywhere then I guess we would want to pass the whole checkoutSession object in case we needed it in the future WDYT?

File: example-app/src/main/java/com/adyen/checkout/example/data/api/model/SessionRequest.kt
Comment (OscarSpruit) [reply example-app/src/main/java/com/adyen/checkout/example/data/api/model/SessionRequest.kt:31]: Oh okay I see 👍

File: sessions-core/src/main/java/com/adyen/checkout/sessions/model/setup/SessionSetupResponse.kt
Comment (etman55) [reply sessions-core/src/main/java/com/adyen/checkout/sessions/model/setup/SessionSetupResponse.kt:72]: Do we use the `serialize` method for responses? I mean it's nice to have both, but do we actually use it?

File: bcmc/src/test/java/com/adyen/checkout/bcmc/BcmcComponentParamsMapperTest.kt
Comment (OscarSpruit) [reply bcmc/src/test/java/com/adyen/checkout/bcmc/BcmcComponentParamsMapperTest.kt:174]: As you said, the test method makes it clear what each boolean means. IMO we could also just add a little comment here to tell what each boolean means

Comment (etman55) [reply bcmc/src/test/java/com/adyen/checkout/bcmc/BcmcComponentParamsMapperTest.kt:174]: I agree with @OscarSpruit I guess just a comment should be enough.

File: drop-in/src/main/java/com/adyen/checkout/dropin/ui/viewmodel/DropInSavedStateHandleContainer.kt
Comment (etman55) [reply drop-in/src/main/java/com/adyen/checkout/dropin/ui/viewmodel/DropInSavedStateHandleContainer.kt:79]: I agree with you, I am just concerned that we might need something else inside the `ComponentParams` not just  `SessionSetupConfiguration`

File: sessions-core/src/main/java/com/adyen/checkout/sessions/model/setup/SessionSetupResponse.kt
Comment (jreij) [reply sessions-core/src/main/java/com/adyen/checkout/sessions/model/setup/SessionSetupResponse.kt:72]: We don't use it but for the sake of consistency let's do it, since we serialize the rest of the fields

File: drop-in/src/main/java/com/adyen/checkout/dropin/ComponentParsingProvider.kt
Comment (jreij) [reply drop-in/src/main/java/com/adyen/checkout/dropin/ComponentParsingProvider.kt:424]: For now there is nothing from the configuration that is required for the stored component, but if we add later it'll be a simple change

File: drop-in/src/main/java/com/adyen/checkout/dropin/ui/viewmodel/DropInSavedStateHandleContainer.kt
Comment (jreij) [reply drop-in/src/main/java/com/adyen/checkout/dropin/ui/viewmodel/DropInSavedStateHandleContainer.kt:79]: Yeah I agree but it seems like the API only wants to use this object to configure components, I guess we'll refactor it if that changes 🤷

File: drop-in/src/main/java/com/adyen/checkout/dropin/ComponentParsingProvider.kt
Comment (jreij) [reply drop-in/src/main/java/com/adyen/checkout/dropin/ComponentParsingProvider.kt:404]: Yes indeed let's pass the same param everywhere (whether it's `CheckoutSession` or `SessionSetupConfiguration`) to keep providers consistent

File: drop-in/src/main/java/com/adyen/checkout/dropin/ComponentParsingProvider.kt
Comment (jreij) [inline drop-in/src/main/java/com/adyen/checkout/dropin/ComponentParsingProvider.kt:345]: Let's remove the default value here so we don't forget to pass `sessionSetupConfiguration` whenever this method is used

File: drop-in/src/main/java/com/adyen/checkout/dropin/ComponentParsingProvider.kt
Comment (jreij) [inline drop-in/src/main/java/com/adyen/checkout/dropin/ComponentParsingProvider.kt:389]: Same here with the default value

File: bcmc/src/test/java/com/adyen/checkout/bcmc/BcmcComponentParamsMapperTest.kt
Comment (jreij) [inline bcmc/src/test/java/com/adyen/checkout/bcmc/BcmcComponentParamsMapperTest.kt:97]: Hmm this is strange, we shouldn't get a `MaxLineLength` error when a method is between ` `` `

Comment (etman55) [reply bcmc/src/test/java/com/adyen/checkout/bcmc/BcmcComponentParamsMapperTest.kt:97]: It's strange, I added it after running `./gradlew check` and it gave me an error!

File: example-app/src/main/java/com/adyen/checkout/example/ui/card/SessionsCardTakenOverViewModel.kt
Comment (jreij) [inline example-app/src/main/java/com/adyen/checkout/example/ui/card/SessionsCardTakenOverViewModel.kt:116]: Minor: this change is unnecessary

File: example-app/src/main/java/com/adyen/checkout/example/ui/card/SessionsCardViewModel.kt
Comment (jreij) [inline example-app/src/main/java/com/adyen/checkout/example/ui/card/SessionsCardViewModel.kt:102]: Minor: this change is unnecessary

File: sessions-core/src/main/java/com/adyen/checkout/sessions/CheckoutSession.kt
Comment (jreij) [inline sessions-core/src/main/java/com/adyen/checkout/sessions/CheckoutSession.kt:22]: Is this still needed?

**Comment by jreij** [comment]
LGTM 👍  We just need to wait until the other PRs are merged then rebase and apply the same changes to the new components

**Review by jreij — APPROVED** [review]

**Pull request merged**
